e2e/testcases/sync_ordering_test.go

```diff
@@ -662,6 +666,114 @@ func TestDependencyWithReconciliation(t *testing.T) {
+	ns1 := func() *corev1.Namespace {
+		return k8sobjects.NamespaceObject("ns1")
+	}
+	anvilCRD := func() *apiextensionsv1.CustomResourceDefinition {
+		return anvilV1CRD()
+	}
+	anvilCR := func() *unstructured.Unstructured {
+		o := newAnvilObject("v1", "anvil", 42)
+		o.SetNamespace("foo")
+		return o
+	}
+	tenantNS1 := "tenant-ns-1"
+	tenantNS2 := "tenant-ns-2"
+	cm1 := func() *corev1.ConfigMap {
+		return k8sobjects.ConfigMapObject(core.Name("cm1"), core.Namespace(tenantNS1))
+	}
+	sa1 := func() *corev1.ServiceAccount {
+		return k8sobjects.ServiceAccountObject("sa1", core.Namespace(tenantNS2))
+	}
+	rootSync := nomostest.DefaultRootSyncID
+	rootSync1 := core.RootSyncID("root-sync-1")
+	rootSync2 := core.RootSyncID("root-sync-2")
+	repoSync1 := core.RepoSyncID("repo-sync-1", tenantNS1)
+	repoSync2 := core.RepoSyncID("repo-sync-2", tenantNS2)
+	nt := nomostest.New(t, nomostesting.Lifecycle,
+		ntopts.SyncWithGitSource(rootSync, ntopts.Unstructured),
+		ntopts.SyncWithGitSource(rootSync1, ntopts.Unstructured),
+		ntopts.SyncWithGitSource(rootSync2, ntopts.Unstructured),
+		ntopts.SyncWithGitSource(repoSync1, ntopts.Unstructured),
+		ntopts.SyncWithGitSource(repoSync2, ntopts.Unstructured),
+		ntopts.WithDelegatedControl,
+		ntopts.WithDeletionPropagationPolicy(metadata.DeletionPropagationPolicyOrphan),
+		ntopts.RepoSyncPermissions(policy.CoreAdmin()))
+
+	nt.T.Log("Add all resources to a single RootSync")
+	rootSyncGitRepo := nt.SyncSourceGitReadWriteRepository(rootSync)
+	nt.Must(rootSyncGitRepo.Add("acme/ns1.yaml", ns1()))
+	nt.Must(rootSyncGitRepo.Add("acme/anvil_crd.yaml", anvilCRD()))
+	nt.Must(rootSyncGitRepo.Add("acme/anvil_cr.yaml", anvilCR()))
+	nt.Must(rootSyncGitRepo.Add("acme/cm1.yaml", cm1()))
+	nt.Must(rootSyncGitRepo.Add("acme/sa1.yaml", sa1()))
+	nt.Must(rootSyncGitRepo.CommitAndPush("Add objects"))
+	nt.Must(nt.WatchForAllSyncs())
+	// Validate metadata for each object
+	nt.Must(verifyManagedBy(nt, ns1(), rootSync))
+	nt.Must(verifyManagedBy(nt, anvilCRD(), rootSync))
+	nt.Must(verifyManagedBy(nt, anvilCR(), rootSync))
+	nt.Must(verifyManagedBy(nt, cm1(), rootSync))
+	nt.Must(verifyManagedBy(nt, sa1(), rootSync))
+
+	nt.T.Cleanup(func() {
+		rs := nomostest.RootSyncObjectV1Beta1FromRootRepo(nt, rootSync.Name)
+		nt.Must(nt.KubeClient.Apply(rs))
+		nt.Must(nt.WatchForAllSyncs())
+	})
+	nt.T.Log("Delete the single RootSync to orphan all resources")
+	nt.Must(nt.KubeClient.Delete(k8sobjects.RootSyncV1Beta1(rootSync.Name)))
+	nt.Must(nt.Watcher.WatchForNotFound(kinds.RootSyncV1Beta1(), rootSync.Name, rootSync.Namespace))
+	nt.Must(verifyUnmanaged(nt, ns1()))
+	nt.Must(verifyUnmanaged(nt, anvilCRD()))
+	nt.Must(verifyUnmanaged(nt, anvilCR()))
+	nt.Must(verifyUnmanaged(nt, cm1()))
+	nt.Must(verifyUnmanaged(nt, sa1()))
+
+	nt.T.Cleanup(func() {
+		// Remove objects from the original RSync for cleanup
+		repo := nt.SyncSourceGitReadWriteRepository(rootSync)
+		nt.Must(repo.Remove("acme/ns1.yaml"))
```

Comment:
Might be nice to have an nt.Should that calls nt.T.Error, instead of nt.T.Fatal. Otherwise the first that fails will fail to clean the rest up.

But that would need a different PR.